### DISCUSSION TITLE
Adding DaisyUI makes all text gray? How to fix?

### DISCUSSION THREAD

**Discussion by AVGVSTVS96** [saadeghi/daisyui#2877] [Q&A]
After adding DaisyUI to my Astro/TW/Typography project, almost all of my light mode text is now a muted light gray instead of the default black. This seems to apply for all text that was black by default. Dark mode colors are not affected, I assume because I already defined dark mode classes in my code.

DaisyUI also added border-radius to prose-code blocks. How can I disable this and use my originally defined border radius of rounded-md?

**Comment by saadeghi** [comment]
Please share a reproduction link

**Comment by ludson96** [comment]
I don't know if you've already solved it, but you should disable all themes, change your file `tailwind.config.js`:

```
module.exports = {
  //...
  daisyui: {
    themes: [],
  },
}
```

**Reply by OneTuskedMario** [reply]
adding this breaks all the input components and I no longer am able to interact with them

**Comment by lihsai0** [comment]
@saadeghi 

I think this is about the theme color of *DaisyUI* overwrites the `--tw-prose-*`. The *tailwindcss-typography* provides [multiple gray scales](https://github.com/tailwindlabs/tailwindcss-typography?tab=readme-ov-file#choosing-a-gray-scale) to choose. Should we provide this in *DaisyUI* too? or just reduce the *DaisyUI* selector weight to make `prose-*` work again?

https://play.tailwindcss.com/TBr3W8GMC1

**Reply by saadeghi** [reply]
The color provided by Typography plugin are hardcoded colors (for example text color = black) and that's not what you would want with daisyUI because if text color is always black, it looks good on a light theme but it would be unreadable on a dark theme. So daisyUI overrides those colors so it will adapt the current theme no matter what.

**Comment by mohamedhekal** [comment]
Here’s how to fix both problems:

---

### ✅ **1. Light Mode Text Turned Gray (Instead of Black)**

DaisyUI applies base styles using the `theme` configuration, including text color.

#### 🔧 **Fix**: Override the default `text-base-content` color

Add this to your `tailwind.config.js`:

```js
module.exports = {
  daisyui: {
    themes: [
      {
        mytheme: {
          ...require("daisyui/src/colors/themes")["[data-theme=light]"],
          "base-content": "#000000", // Set default text color to black
        },
      },
    ],
  },
  theme: {
    extend: {},
  },
  plugins: [require("daisyui"), require("@tailwindcss/typography")],
}
```

Then in your HTML, make sure you're using your theme:

```html
<html data-theme="mytheme">
```

Alternatively, **remove `base-content` inheritance** entirely by explicitly styling your body or typography classes.

---

### ✅ **2. Unwanted Border-Radius on `<code>` blocks in `prose`**

DaisyUI adds `rounded-box` by default to many elements, including `code`.

#### 🔧 **Fix**: Override `prose code` styles in your CSS

Add this in your global CSS or layout styles:

```css
.prose code {
  @apply rounded-md; /* Your preferred radius */
}
```

If DaisyUI is still applying `rounded-box`, you can override it like this:

```css
.prose code {
  @apply !rounded-md; /* Force override */
}
```

Or disable all DaisyUI component styles on prose if needed (via custom class exclusions).

---

### ✅ Bonus: Disable DaisyUI’s Global Styles (Optional)

If DaisyUI’s styles are too opinionated, you can **disable its base styles**:

```js
daisyui: {
  styled: false, // disables all default component styles
}
```

But this will remove all DaisyUI component theming too — use carefully.

---
